docs/api/fiber.md

```diff
@@ -205,7 +205,7 @@ Shutdown gracefully shuts down the server without interrupting any active connec
 
 ShutdownWithTimeout will forcefully close any active connections after the timeout expires.
 
-ShutdownWithContext shuts down the server including by force if the context's deadline is exceeded.
+ShutdownWithContext shuts down the server including by force if the context's deadline is exceeded. Shutdown hooks will still be executed, even if an error occurs during the shutdown process, as they are deferred to ensure cleanup happens regardless of errors.
```

Comment:
Not sure about this behavior. I think we should only execute them during successful shutdown. Maybe we can OnShutdownError hook. What do you think @gaby @ReneWerner87

Reply:
I'm not sure either. Also not sure about the defer of the hook

Reply:
I would prefer to add Post Pre hook

Reply:
The add Post Pre hook is a good idea, but how do I tell which hooks are pre closed and which are post closed? @efectn

Reply:
You can add new hook called PostShutdown, PreShutdown and replace the old one with them

Reply:
sound good, @JIeJaitt can you do the suggested change

Reply:
This means making a change below the `type Hooks struct` to add these two fields that mean `PreShutdownHandler` and `PostShutdownHandler` respectively , if I understand it correctly I'll try to implement it @ReneWerner87 @efectn @gaby

Reply:
> This means making a change below the `type Hooks struct` to add these two fields that mean `PreShutdownHandler` and `PostShutdownHandler` respectively , if I understand it correctly I'll try to implement it @ReneWerner87 @efectn @gaby

Yes it is. You can also add err parameter to PostShutdown hook, so that we can also get rid of OnShutdownError, OnShutdownSuccess properties of ListenerConfig.

Reply:
> > This means making a change below the `type Hooks struct` to add these two fields that mean `PreShutdownHandler` and `PostShutdownHandler` respectively , if I understand it correctly I'll try to implement it @ReneWerner87 @efectn @gaby
> 
> Yes it is. You can also add err parameter to PostShutdown hook, so that we can also get rid of OnShutdownError, OnShutdownSuccess properties of ListenerConfig.

OK, I will press on with this idea to realize it again in my free time, by the way what is the v3 deadline as I am busy with my final exams these days.

Reply:
> > > This means making a change below the `type Hooks struct` to add these two fields that mean `PreShutdownHandler` and `PostShutdownHandler` respectively , if I understand it correctly I'll try to implement it @ReneWerner87 @efectn @gaby
> > 
> > 
> > Yes it is. You can also add err parameter to PostShutdown hook, so that we can also get rid of OnShutdownError, OnShutdownSuccess properties of ListenerConfig.
> 
> OK, I will press on with this idea to realize it again in my free time, by the way what is the v3 deadline as I am busy with my final exams these days.

We want to release v3 rc1 in the beginning of January. If you can't look at the PR this week, i can also look no problem